### PR TITLE
Create 2024-12-11-security-audit.md

### PR DESCRIPTION
**Pull request by rhiannon-io**

### CODE REVIEW

File: _posts/2024-12-11-security-audit.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+These gaps could create vulnerabilities if left unaddressed, but Trail of Bits provided specific recommendations for us to mitigate risks. For example, they highlighted the need for better isolation between AWS accounts and more granular access control policies. These findings have already informed a project proposal we’re developing to modernize our infrastructure and automate resource management through tools like Terraform.
+
+One of the most valuable outcomes on the infrastructure side was the clarity it brought in helping us better prioritize our efforts next year. Our team now has a roadmap for what improvements we need to make, how to implement them, and the expected impact on security.
+
+# Next steps for RubyGems.org security
+
+**We are using the audit’s findings to shape the next steps for security updates on RubyGems.org:**
+
+**- Short-term:** We’ve already implemented many small, immediate improvements, such as stricter TLS enforcement and better access controls.
```
Comment (deivid-rodriguez) [inline _posts/2024-12-11-security-audit.md:55]: I think this is how the list would render correctly, and similarly with the other bullet points.

```suggestion
- **Short-term:** We’ve already implemented many small, immediate improvements, such as stricter TLS enforcement and better access controls.
```

**Review by deivid-rodriguez — APPROVED** [review]
Looks good to me! Just found a potential and minor rendering issue.

**Review by martinemde — COMMENTED** [review]
The article looks good. Just need the markdown fixes that @deivid-rodriguez suggested and we can merge.

File: _posts/2024-12-11-security-audit.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+author_email: rhiannon@rubycentral.org
+---
+
+At Ruby Central, ensuring the security of RubyGems.org—the central hub for Ruby packages—is one of our top priorities. With over 184 billion downloads, RubyGems.org is crucial to the Ruby ecosystem, supporting developers, businesses, and open source projects worldwide. 
+
+To strengthen this critical infrastructure, we recently collaborated with [Trail of Bits](https://www.trailofbits.com/) on our first security audit of RubyGems.org.
+
+Trail of Bits is a leading cybersecurity firm that helps organizations identify vulnerabilities, enhance defenses, and implement security best practices. 
+
+The audit was funded through a grant from the Alpha-Omega Project, a consortium backed by Microsoft, Google, and Amazon, dedicated to improving the security of open source projects.
```
Comment (segiddins) [inline _posts/2024-12-11-security-audit.md:14]:
```suggestion
The audit was funded through a grant from the [Alpha-Omega Project](https://alpha-omega.dev), a consortium backed by Microsoft, Google, and Amazon, dedicated to improving the security of open source projects.
```

**Review by martinemde — APPROVED** [review]

File: _posts/2024-12-11-security-audit.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+Trail of Bits leveraged advanced tools like Semgrep for static analysis, Burp Suite Professional for dynamic testing, and Ruzzy for fuzzing critical components, ensuring a thorough and multi-layered audit. 
+
+The audit identified 33 issues, including seven medium-severity items and one high-severity item. **Notably, most of these findings do not constitute actual security breaches. Our team has been addressing each finding and using these insights to bolster RubyGems.org’s security posture.**
+
+## RubyGems.org Rails app
+
+Trail of Bits reviewed the Rails application code extensively and found that, in general, the site is built securely. Many of the issues highlighted were minor adjustments, often related to default Rails configurations that could be made more secure.
+
+For example, one notable finding involved a vulnerability in the email system, where fallback to unencrypted transmission could expose sensitive emails. The recommended fix was straightforward—replace enable_starttls_auto with enable_starttls to enforce strict TLS encryption.
```
Comment (segiddins) [inline _posts/2024-12-11-security-audit.md:33]:
```suggestion
For example, one notable finding involved a vulnerability in the email system, where fallback to unencrypted transmission could expose sensitive emails. The recommended fix was straightforward—replace `enable_starttls_auto` with `enable_starttls` to enforce strict TLS encryption.
```

**Review by segiddins — APPROVED** [review]

**Pull request merged**